ScarletEngine/src/Core/Logger.cpp

```diff
@@ -0,0 +1,77 @@
+#include "Core/Logger.h"
+
+namespace ScarletEngine
+{
+	Logger::~Logger()
+	{
+		if (LogFile)
+		{
+			fclose(LogFile);
+		}
+	}
+
+	void Logger::Log(LogLevel Level, const char* Message) const
+	{
+		static const char* VerbosePrefix = "[Verbose] ";
+		static const char* InfoPrefix = "[Info] ";
+		static const char* WarningPrefix = "[Warn] ";
+		static const char* ErrorPrefix = "[Error] ";
+		static const char* DefaultPrefix = "";
+
+		const char** PrefixPtr = nullptr;
+		switch (Level)
+		{
+		case LogLevel::LogVerbose:
+			PrefixPtr = &VerbosePrefix;
+			break;
+		case LogLevel::LogInfo:
+			PrefixPtr = &InfoPrefix;
+			break;
+		case LogLevel::LogWarning:
+			PrefixPtr = &WarningPrefix;
+			break;
+		case LogLevel::LogError:
+			PrefixPtr = &ErrorPrefix;
+			break;
+		default:
+			PrefixPtr = &DefaultPrefix;
+			break;
+		}
+
+		printf("%s", *PrefixPtr);
+		printf("%s", Message);
+		printf("\n");
+
+		// If compiling for Windows and in debug mode, 
+		// we can print to the debug output (visible in Visual Studio Output tab)
+#if defined(_WIN32) && defined(DEBUG)
+		OutputDebugString(*PrefixPtr);
+		OutputDebugString(Message);
+		OutputDebugString("\n");
+#endif
+		if (LogFile)
+		{
+			time_t Now = time(0);
+			struct tm* TimeStruct;
+			char Buffer[80];
+			TimeStruct = localtime(&Now);
+
+			assert(TimeStruct != nullptr);
+			strftime(Buffer, sizeof(Buffer), "[%Y-%m-%d.%X] ", TimeStruct);
+
+			fprintf(LogFile, "%s", Buffer);
```

Comment:
would using std::fstream be faster? Something like I did here on another project: https://github.com/filrod/ComputerGraphicsLearning_PublicVersion/blob/908dcce70e3d7b8a6b46fee9c4425e751d864b77/RayTracerInAWeekend/RayTracerInAWeekend/FileHandler.h#L18

This is minor and I'm just wondering so I will merge

Reply:
It's mostly just a style thing. `fstream` is very C++ but since we're using c strings and format strings rather than `std::string`s I figured I might as well just follow the C style for this whole logging system.

Reply:
I was mostly just trying to avoid using `std::string` as much as possible and stick to string literals and a stack-based buffer rather than dynamic heap allocation (which comes with c++ strings)